### PR TITLE
pythonPackages.openstack: init

### PR DESCRIPTION
**Pull request by superherointj**
Initial port of OpenStack packages (25 to start).

---

Tests are disabled because they are too tricky to get right. I managed to enable tests on several packages but as they have cyclic dependencies between them that couldn't get around by the usual workarounds I've come to disable them all (and I've dropped all the tests code). I don't plan spending anymore time on getting tests enabled here. All packages and applications seem to be working fine. It's just tests really.

---

To build:
```
nix build \
  .#python3Packages.openstack-debtcollector \
  .#python3Packages.openstack-docstheme \
  .#python3Packages.openstack-hacking \
  .#python3Packages.openstack-keystoneauth1 \
  .#python3Packages.openstack-os-client-config \
  .#python3Packages.openstack-os-service-types \
  .#python3Packages.openstack-osc-lib \
  .#python3Packages.openstack-oslo_concurrency \
  .#python3Packages.openstack-oslo_config \
  .#python3Packages.openstack-oslo_context \
  .#python3Packages.openstack-oslo_i18n \
  .#python3Packages.openstack-oslo_log \
  .#python3Packages.openstack-oslo_serialization \
  .#python3Packages.openstack-oslo_utils \
  .#python3Packages.openstack-oslotest \
  .#python3Packages.openstack-osprofiler \
  .#python3Packages.openstack-pbr \
  .#python3Packages.openstack-python-cinderclient \
  .#python3Packages.openstack-python-client \
  .#python3Packages.openstack-python-keystoneclient \
  .#python3Packages.openstack-python-manilaclient \
  .#python3Packages.openstack-python-novaclient \
  .#python3Packages.openstack-sdk \
  .#python3Packages.openstack-tempest
```

### CODE REVIEW

**Review by mweinelt — CHANGES_REQUESTED** [review]
This needs to be split into one commit per package added.

File: pkgs/development/python-modules/pbr/default.nix [hunk 1]
```diff
@@ -1,20 +0,0 @@
...
-
-  # circular dependencies with fixtures
-  doCheck = false;
-
-  meta = with lib; {
-    homepage = "http://docs.openstack.org/developer/pbr/";
-    license = licenses.asl20;
-    description = "Python Build Reasonableness";
-  };
-}
```
Comment (mweinelt) [inline pkgs/development/python-modules/pbr/default.nix:20]: Removing/replacing the package needs an explanation.

Comment (superherointj) [reply pkgs/development/python-modules/pbr/default.nix:20]: `pbr` is part of Openstack. Explanation where?

**Comment by superherointj** [comment]
> This needs to be split into one commit per package added.

Last time I splited everything it was squashed in the merge. So it served no purpose. If it is not squashed I'll promptly do so.

**Comment by mweinelt** [comment]
I will not squash your commits, in fact nobody should. One commit per package stems from the commit message requirement and everybody should follow that.

**Comment by mweinelt** [comment]
I think we should begin with getting both stestr and openstack-hacking into #136470 and solve the infinite recursion situation and then build upon that.

Since python breaks so easily I would strongly recommend to setup a proper check phase, so we run the upstream test suites.

**Comment by superherointj** [comment]
> I think we should begin with getting both stestr and openstack-hacking into #136470 and solve the infinite recursion situation and then build upon that.
> 
> Since python breaks so easily I would strongly recommend to setup a proper check phase on every package, so we run the upstream test suites, if any exists.

I could not find a proper solution to have tests working in all packages. In some packages it will work fine. But when adding tests to all packages it just loops. Despite using the usual workarounds (`overridePythonAttrs` and `passthru.tests`). There is a lot of cross-references between packages here. More than usual I guess.

**Comment by mweinelt** [comment]
Then testing the packages that hold the more important functionality should be preferred.

**Comment by fabaff** [comment]
Isn't this a duplicate of #134282 or a partial overlap?

**Comment by superherointj** [comment]
> Isn't this a duplicate of #134282 or a partial overlap?

Wasn't aware. Odd that I searched and I found one that was closed. I guess it is a partial overlap then. I only need the clients.

**Pull request closed without merging**